code/game/machinery/computer/buildandrepair.dm

```diff
@@ -389,94 +432,127 @@
 		return
 	return ..()
 
-/obj/item/circuitboard/rdconsole/attackby(obj/item/I as obj, mob/user as mob, params)
-	if(istype(I,/obj/item/card/id)||istype(I, /obj/item/pda))
+/obj/item/circuitboard/rdconsole/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/card/id) || istype(I, /obj/item/pda))
 		if(allowed(user))
 			user.visible_message("<span class='notice'>\the [user] waves [user.p_their()] ID past the [src]'s access protocol scanner.</span>", "<span class='notice'>You swipe your ID past the [src]'s access protocol scanner.</span>")
 			var/console_choice = input(user, "What do you want to configure the access to?", "Access Modification", "R&D Core") as null|anything in access_types
-			if(console_choice == null)
+			if(!console_choice)
 				return
 			switch(console_choice)
 				if("R&D Core")
-					name = "Circuit Board (RD Console)"
+					board_name = "RD Console"
 					build_path = /obj/machinery/computer/rdconsole/core
 					id = 1
 				if("Robotics")
-					name = "Circuit Board (RD Console - Robotics)"
+					board_name = "RD Console - Robotics"
 					build_path = /obj/machinery/computer/rdconsole/robotics
 					id = 2
 				if("E.X.P.E.R.I-MENTOR")
-					name = "Circuit Board (RD Console - E.X.P.E.R.I-MENTOR)"
+					board_name = "RD Console - E.X.P.E.R.I-MENTOR"
 					build_path = /obj/machinery/computer/rdconsole/experiment
 					id = 3
 				if("Mechanics")
-					name = "Circuit Board (RD Console - Mechanics)"
+					board_name = "RD Console - Mechanics"
 					build_path = /obj/machinery/computer/rdconsole/mechanics
 					id = 4
 				if("Public")
-					name = "Circuit Board (RD Console - Public)"
+					board_name = "RD Console - Public"
 					build_path = /obj/machinery/computer/rdconsole/public
 					id = 5
 
+			format_board_name()
 			to_chat(user, "<span class='notice'>Access protocols set to [console_choice].</span>")
 		else
 			to_chat(user, "<span class='warning'>Access Denied</span>")
 		return
 	return ..()
 
-/obj/structure/computerframe/attackby(obj/item/P as obj, mob/user as mob, params)
+
+/obj/structure/computerframe
+	name = "computer frame"
+	icon = 'icons/obj/stock_parts.dmi'
+	icon_state = "0"
+	density = TRUE
+	max_integrity = 100
+	var/state = 0
+	var/obj/item/circuitboard/circuit = null
+	var/base_mineral = /obj/item/stack/sheet/metal
+
+/obj/structure/computerframe/deconstruct(disassembled = TRUE)
+	if(!(flags & NODECONSTRUCT))
+		drop_computer_parts()
+	return ..() // will qdel the frame
+
+/obj/structure/computerframe/obj_break(damage_flag)
+	deconstruct()
+
+/obj/structure/computerframe/proc/drop_computer_parts()
+	new base_mineral(loc, 5)
+	if(circuit)
+		circuit.forceMove(loc)
+		circuit = null
+	if(state >= 3)
+		var/obj/item/stack/cable_coil/A = new /obj/item/stack/cable_coil(loc)
+		A.amount = 5
+	if(state >= 4)
+		new /obj/item/stack/sheet/glass(loc, 2)
+
+/obj/structure/computerframe/attackby(obj/item/I, mob/user, params)
```

Comment:
Christ. Lots of tool behaviour in here ... could you rewrite those to `XXX_act`s if you have time spare?